src/lib/allocation_tracker.cc

```diff
@@ -100,45 +101,49 @@ DDRes AllocationTracker::allocation_tracking_init(
     DDRES_RETURN_ERROR_LOG(DD_WHAT_UKNW, "Allocation profiler already started");
   }
 
-  DDRES_CHECK_FWD(instance->init(allocation_profiling_rate,
-                                 flags & kDeterministicSampling,
-                                 flags & kTrackDeallocations, stack_sample_size,
-                                 ring_buffer, timer_check));
+  DDRES_CHECK_FWD(instance->init(allocation_profiling_rate, flags,
+                                 stack_sample_size, ring_buffer, timer_check));
   _instance = instance;
 
   state.init(true, flags & kTrackDeallocations);
 
   return {};
 }
 
-DDRes AllocationTracker::init(uint64_t mem_profile_interval,
-                              bool deterministic_sampling,
-                              bool track_deallocations,
+DDRes AllocationTracker::init(uint64_t mem_profile_interval, uint32_t flags,
                               uint32_t stack_sample_size,
                               const RingBufferInfo &ring_buffer,
                               const IntervalTimerCheck &timer_check) {
   _sampling_interval = mem_profile_interval;
-  _deterministic_sampling = deterministic_sampling;
+  _deterministic_sampling = flags & kDeterministicSampling;
   _stack_sample_size = stack_sample_size;
-  if (ring_buffer.ring_buffer_type !=
-      static_cast<int>(RingBufferType::kMPSCRingBuffer)) {
-    return ddres_error(DD_WHAT_PERFRB);
+  _otel_profiler_mode = flags & kOtelProfilerMode;
+
+  if (!_otel_profiler_mode) {
+    if (ring_buffer.ring_buffer_type !=
+        static_cast<int>(RingBufferType::kMPSCRingBuffer)) {
+      return ddres_error(DD_WHAT_PERFRB);
+    }
+
+    DDRES_CHECK_FWD(ddprof::ring_buffer_attach(ring_buffer, &_pevent));
+
+    const auto &rb = _pevent.rb;
+    if (rb.tsc_available) {
+      TscClock::init(TscClock::CalibrationParams{
+          .offset = TscClock::time_point{TscClock::duration{rb.time_zero}},
+          .mult = rb.time_mult,
+          .shift = rb.time_shift});
+    }
+    PerfClock::init(static_cast<PerfClockSource>(rb.perf_clock_source));
+  } else {
+    PerfClock::init(PerfClockSource::kClockMonotonic);
```

Comment:
because monotonic is what the full host profiler would expect, and tsc logic is not available

Reply:
On otel profiler mode, `PerfClock` is not used to timestamp samples(since it's done on eBFP side), I had to initialize it because it's also used to determine when periodically check for newly loaded libraries.